src/webpub_manifest_parser/core/ast.py

```diff
@@ -0,0 +1,831 @@
+            (self.name, self.sort_as, self.code, self.scheme, frozenset(self.links))
+        )
+
+    def __repr__(self):
+        """Return a string representation of the object.
+
+        :return: String representation
+        :rtype: str
+        """
+        return (
+            u"<Subject(name={0}, sort_as={1}, code={2}, scheme={3}, links={4})>".format(
+                self.name, self.sort_as, self.code, self.scheme, self.links
+            )
+        )
+
+
+class SubjectProperty(ParsableProperty):
+    """Property containing information about a subject."""
+
+    PARSER = AnyOfParser(
+        [
+            StringParser(),
+            ArrayParser(AnyOfParser([StringParser(), TypeParser(Subject)])),
+            TypeParser(Subject),
+        ]
+    )
+
+
+class Owner(Node, PropertiesGrouping):
+    """Object containing information about the collection's owners."""
+
+    collection = TypeProperty("collection", required=False, nested_type=Contributor)
+    series = TypeProperty("series", required=False, nested_type=Contributor)
+
+    def __hash__(self):
+        """Calculate the hash.
+
+        :return: Hash
+        :rtype: int
+        """
+        return hash((self.collection, self.series))
+
+    def __repr__(self):
+        """Return a string representation of the object.
+
+        :return: String representation
+        :rtype: str
+        """
+        return u"<Owner(collection={0}, series={1})>".format(
+            self.collection, self.series
+        )
+
+
+class Metadata(Node):
+    """Dictionary containing manifest's metadata."""
+
+    identifier = URIProperty("identifier", required=False)
+    type = URIProperty("@type", required=False)
+    title = LocalizableStringProperty("title", required=True)
+    subtitle = LocalizableStringProperty("subtitle", required=False)
+    modified = DateTimeProperty("modified", required=False)
+    published = DateOrTimeProperty("published", required=False)
+    language = LanguageProperty("language", required=False)
+    sort_as = StringProperty("sortAs", required=False)
+    author = ContributorProperty("author", required=False)
+    translator = ContributorProperty("translator", required=False)
+    editor = ContributorProperty("editor", required=False)
+    artist = ContributorProperty("artist", required=False)
+    illustrator = ContributorProperty("illustrator", required=False)
+    letterer = ContributorProperty("letterer", required=False)
+    penciler = ContributorProperty("penciler", required=False)
+    colorist = ContributorProperty("colorist", required=False)
+    inker = ContributorProperty("inker", required=False)
+    narrator = ContributorProperty("narrator", required=False)
+    contributor = ContributorProperty("contributor", required=False)
+    publisher = ContributorProperty("publisher", required=False)
+    imprint = ContributorProperty("imprint", required=False)
+    subject = SubjectProperty("subject", required=False)
+    reading_progression = EnumProperty(
```

Comment:
The `reading_progression` is a hint but is not very useful for the navigator side, so [we added an `effectiveReadingProgression` helper](https://github.com/readium/architecture/blob/master/proposals/004-publication-helpers-services.md#presentation) to calculate the actual reading progression: 

[In this Kotlin commit](https://github.com/readium/r2-shared-kotlin/pull/126/commits/3e76508c5778fae2d3f3dbe6df6fd4b99b789a48) you can find an example implementation with a set of test cases.

Although this is only useful if this AST is meant to be used to render publication.